app/controllers/tweets_controller.rb

```diff
@@ -0,0 +1,70 @@
+class TweetsController < ApplicationController
+  get '/tweets' do
+    if !logged_in?
+      redirect "/login"
+    else
+      @tweets = Tweet.all
+      erb :'/tweets/tweets'
+    end
+  end
+
+  get '/tweets/new' do
+    if !logged_in?
+      redirect "/login"
+    else
+      erb :'/tweets/create_tweet'
+    end
+  end
+
+  post '/tweets' do
+    if params[:content].strip != ""
+      @tweet = current_user.tweets.create(content: params[:content])
+      redirect "/tweets/#{@tweet.id}"
+    else
+      redirect "/tweets/new"
+    end
+  end
+
+  get '/tweets/:id' do
+    if !logged_in?
+      redirect "/login"
+    else
+      @tweet = Tweet.find_by(params[:id])
+      erb :'/tweets/show_tweet'
+    end
+  end
+
+  get '/tweets/:id/edit' do
+    if !logged_in?
+      redirect "/login"
+    else
+      @tweet = Tweet.find_by(params[:id])
+      if @tweet.user_id == current_user.id
+        erb :'/tweets/edit_tweet'
+      else
+        redirect "/tweets"
+      end
+    end
+  end
+
+  patch '/tweets/:id' do
+    if params[:content] != ""
```

Comment:
I would contend that you still want to double check the user's association to the tweet here. There are other ways to submit a PATCH/POST request than going through an HTML form...